src/main/java/Duke.java

```diff
@@ -1,10 +1,177 @@
+            } else {
+                Task n = new Task(input);
+                AL.add(n);
+                System.out.println("    ____________________________________________________________");
+                System.out.println("    added: " + input);
+                System.out.println("    ____________________________________________________________");
+            }
+            saveTasksToFile(AL, FILE_PATH);
+        }
+
+        System.out.println("    ____________________________________________________________");
+        System.out.println("    Bye. Hope to see you again soon!");
+        System.out.println("    ____________________________________________________________");
+    }
+    public static void saveTasksToFile(ArrayList<Task> taskList, String filePath) {
+        try (BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(filePath))) {
+            for (Task task : taskList) {
+                bufferedWriter.write(task.toFileString());
+                bufferedWriter.newLine();
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public static void loadTasksFromFile(ArrayList<Task> taskList, String filePath) {
+        try {
+            FileReader fileReader = new FileReader(filePath);
+            BufferedReader bufferedReader = new BufferedReader(fileReader);
+
+            String line;
+            while ((line = bufferedReader.readLine()) != null) {
+                Task task = createTaskFromLine(line);
+                taskList.add(task);
+            }
+
+            bufferedReader.close();
+        } catch (FileNotFoundException e) {
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private static Task createTaskFromLine(String line) {
+        String[] parts = line.split(" \\| ");
+
+        if(parts.length > 2) {
+            String type = parts[0];
+            String status = parts[1];
+            String description = parts[2];
+
+            Task task;
+
+            switch (type) {
+                case "T":
+                    task = new Todos(description);
+                    break;
+                case "D":
+                    if (parts.length >= 4) {
+                        String by = parts[3];
+                        task = new Deadline(description, by);
+                    } else {
+                        task = new Task(description);
+                    }
+                    break;
+                case "E":
+                    if (parts.length >= 5) {
+                        String from = parts[3];
+                        String to = parts[4];
+                        task = new Events(description, from, to);
+                    } else {
+                        task = new Task(description);
+                    }
+                    break;
+                default:
+                    task = new Task(description);
+            }
+
```

Comment:
It would be better if you have avoided 'deep nesting' within your methods and also the indentation for the switch cases should be on the same as the opening switch statement :))